### ISSUE TITLE
Categories Summary report sub-category total does not match Transaction report sub-category for same period. 

### ISSUE THREAD

**Issue by djmcphie** [moneymanagerex/moneymanagerex#4085]
Hello,

This is my first time submitting an issue, so apologies if I am not doing this correctly.

I have included an excel SS with this submission that details the issue. I have included screen shots and summaries that I hope will illustrate the issue.

Regards
DJ

[MMEx Issue.xlsx](https://github.com/moneymanagerex/moneymanagerex/files/7795346/MMEx.Issue.xlsx)
> Please do not remove information attached below this text.

<hr>Version: 1.5.11 64-bit<br>
 • Build: Dec  7 2021 12:41:07<br>
 • db 7 •  (aes128cbc)<br>
 • git 40b3964 (2021-12-06)<br>
<br>
Libs:<br>
 • wxWidgets 3.1.5 (wxMSW 10.0)<br>
 • wxSQLite3 4.6.0 (SQLite 3.31.1)<br>
 • RapidJSON 1.1.0<br>
 • Lua 5.3.5<br>
 • lunasvg v2.0.1<br>
 • libcurl/7.61.0-DEV WinSSL<br>
 • gettext 0.19.8.1<br>
 • apexcharts.js<br>
<br>
Build using:<br>
 • CMake 3.16.2<br>
 • MSBuild 15.9.21.664<br>
 • MSVS15.0<br>
 • MSVSC  19.16.27035.0<br>
 • Windows SDK 10.0.14393.0<br>
<br>
OS:<br>
 • Windows 10 (build 19043), 64-bit edition<br>
 •  English (U.S.) (windows-1252)<br>
 • 1920x1080 32bit 96x96ppi

**Comment by whalley** [comment]
Think I may have found an issue here, which hopefully fixes the problem.... The way MMEX works with amounts for non-transfers is:

- If Transaction IS NOT SPLIT then amount is stored as +ve amount and then processed accordingly based on DEPOSIT/WITHDRAWAL transaction code.
- If Transaction IS SPLIT and transaction code == DEPOSIT
    - DEPOSIT splits are stored and +ve amount
    - WITHDRAWAL splits are stored as -ve amount
-  If Transaction IS SPLIT and transaction code == WITHDRAWAL
    - WITHDRAWAL splits are stored as +ve amount
    - DEPOSIT splits are stored and -ve amount

The Transaction report processing handles this correctly but the category reports **do not** and results in split transactions being misclassed when accounting for EXPENSES and INCOME.

**Comment by whalley** [comment]
@djmcphie see if this build fixes your issue https://ci.appveyor.com/api/buildjobs/awkxfed7bdgpo3yn/artifacts/mmex-1.5.12-Beta.1-win64.exe

**Comment by djmcphie** [comment]
@whalley , firstly thank you for your speedy work on this issue. I appreciate this very much.
I have updated the program as you suggested.
Transaction report and Summary report now agree for the account tested. (Liabilities:Superannuation payable)
I have also checked all the other Liability accounts and the summary account matches them too.
I have also checked all other asset and income accounts for this period and they match too.
I am satisfied that the issue is now resolved.
Once again, thank you very much.
DJ
